### PR TITLE
[ADD] [16.0] stock_picking_report_delivery_driver

### PR DESCRIPTION
**Pull request by Shide**
Adds Delivery Driver to Stock Picking reports

## Delivery Slip report
image

## Picking Operations report
image

## Delivery Slip without Carrier but Driver
image

MT-6115 @moduon  @rafaelbn @yajo @EmilioPascual @fcvalgar please review if you want :)

### CODE REVIEW

**Review by fcvalgar — APPROVED** [review]
Functional review.

LGTM, Thank you @Shide 

image

Picking Operations 
image

Delivery Slip
image

Only driver
image

Only Carrier
image

File: stock_picking_report_delivery_driver/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+{
+    "name": "Stock Picking Report Delivery Driver",
+    "summary": "Delivery Driver info in Stock Picking reports",
+    "version": "16.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Inventory/Inventory",
+    "website": "https://github.com/OCA/stock-logistics-reporting",
+    "author": "Moduon, Odoo Community Association (OCA)",
+    "maintainers": ["Shide"],
```
Comment (EmilioPascual) [inline stock_picking_report_delivery_driver/__manifest__.py:12]:
```suggestion
    "maintainers": ["Shide", "rafaelbn"],
```

**Review by EmilioPascual — CHANGES_REQUESTED** [review]
Nice Job @Shide. Just a single change

File: stock_picking_report_delivery_driver/report/report_deliveryslip.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        </xpath>
+        <xpath expr="//p[@t-field='o.carrier_id']" position="attributes">
+            <attribute name="t-if">o.carrier_id</attribute>
+            <attribute name="class">mb-0</attribute>
+        </xpath>
+        <xpath expr="//p[@t-field='o.carrier_id']" position="after">
+            <p t-if="o.driver_id">
+                <span t-if="o.carrier_id">(</span>
+                <span t-field="o.driver_id.name" />
+                <span t-if="o.carrier_id">)</span>
```
Comment (yajo) [inline stock_picking_report_delivery_driver/report/report_deliveryslip.xml]: Avoid unneeded whitespace on reports:

```suggestion
                <span t-if="o.carrier_id">(</span><span t-field="o.driver_id.name" /><span t-if="o.carrier_id">)</span>
```

File: stock_picking_report_delivery_driver/report/report_stockpicking_operations.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        </xpath>
+        <xpath expr="//p[@t-field='o.carrier_id']" position="attributes">
+            <attribute name="t-if">o.carrier_id</attribute>
+            <attribute name="class">mb-0</attribute>
+        </xpath>
+        <xpath expr="//p[@t-field='o.carrier_id']" position="after">
+            <p t-if="o.driver_id">
+                <span t-if="o.carrier_id">(</span>
+                <span t-field="o.driver_id.name" />
+                <span t-if="o.carrier_id">)</span>
```
Comment (yajo) [inline stock_picking_report_delivery_driver/report/report_stockpicking_operations.xml]:
```suggestion
                <span t-if="o.carrier_id">(</span><span t-field="o.driver_id.name" /><span t-if="o.carrier_id">)</span>
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Review by EmilioPascual — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]
👍🏼 Thank you!

**Comment by rafaelbn** [comment]
Hello @Shide ! :-)

Test are failing, this is the message:
```

2024-06-02 20:58:05,334 288 ERROR odoo odoo.addons.stock_average_daily_sale.tests.test_average_daily_sale: FAIL: TestAverageSale.test_average_sale
Traceback (most recent call last):
  File "/__w/stock-logistics-reporting/stock-logistics-reporting/stock_average_daily_sale/tests/test_average_daily_sale.py", line 52, in test_average_sale
    self.assertRecordValues(
  File "/opt/odoo/odoo/tests/common.py", line 610, in assertRecordValues
    self.fail('\n'.join(errors))
AssertionError: The records and expected_values do not match.
Wrong number of records to compare: 0 records versus 1 expected values.

==== Missing record ====
{'average_qty_by_sale': 10.0,
 'nbr_sales': 1.0,
 'qty_in_stock': 40.0,
 'recommended_qty': 20.0,
 'warehouse_id': 1}
 
2024-06-02 20:58:05,337 288 INFO odoo odoo.addons.stock_average_daily_sale.tests.test_average_daily_sale: Starting TestAverageSale.test_average_sale_multiple ... 
2024-06-02 20:58:05,631 288 INFO odoo odoo.addons.stock_average_daily_sale.tests.test_average_daily_sale: ====================================================================== 
2024-06-02 20:58:05,631 288 ERROR odoo odoo.addons.stock_average_daily_sale.tests.test_average_daily_sale: FAIL: TestAverageSale.test_average_sale_multiple
Traceback (most recent call last):
  File "/__w/stock-logistics-reporting/stock-logistics-reporting/stock_average_daily_sale/tests/test_average_daily_sale.py", line 129, in test_average_sale_multiple
    self.assertRecordValues(
  File "/opt/odoo/odoo/tests/common.py", line 610, in assertRecordValues
    self.fail('\n'.join(errors))
AssertionError: The records and expected_values do not match.
Wrong number of records to compare: 0 records versus 1 expected values.

==== Missing record ====
{'nbr_sales': 3.0, 'qty_in_stock': 19.0, 'warehouse_id': 1}
```

Could you please take a look? 😄 ❤️ 

Thank you!

**Comment by Shide** [comment]
@rousseldenis @lmignon Could you please check the failing tests in `stock_average_daily_sale` module?

**Comment by rousseldenis** [comment]
> @rousseldenis @lmignon Could you please check the failing tests in `stock_average_daily_sale` module?

@Shide Where ?

**Comment by rousseldenis** [comment]
> > @rousseldenis @lmignon Could you please check the failing tests in `stock_average_daily_sale` module?
> 
> @Shide Where ?

Ok, I see. I can check

**Comment by Shide** [comment]
@rousseldenis what was happening? What do you do to let tests pass?

**Comment by rousseldenis** [comment]
> @rousseldenis what was happening? What do you do to let tests pass?

Apparently, a temporary failure. Wasn't able to reproduce problem locally, I re run the merge

**Pull request merged**